docker/docker-compose.yml

```diff
@@ -1,29 +1,41 @@
+version: "3.7"
+
 services:
   aphrodite-engine:
-    build:
-      context: .
-    restart: on-failure:5
-    environment:
-      - NUM_GPUS=${NUM_GPUS}
-      - MODEL_NAME=${MODEL_NAME}
-      - REVISION=${REVISION}
-      - HUGGING_FACE_HUB_TOKEN=${HUGGING_FACE_HUB_TOKEN}
-      - QUANTIZATION=${QUANTIZATION}
-      - KVCACHE=${KVCACHE}
-      - API_KEY=${API_KEY}
-      - CONTEXT_LENGTH=${CONTEXT_LENGTH}
-      - GPU_MEMORY_UTILIZATION=${GPU_MEMORY_UTILIZATION}
-      - ENFORCE_EAGER=${ENFORCE_EAGER}
-    volumes:
-      - ${HOME}/.cache:/root/.cache/
-    shm_size: 10g
-    ports:
-      - "7860:7860"
-      - "2242:2242"
+    command: --host 0.0.0.0
+             --port 2242
+             ${API_KEY:+--api-keys ${API_KEY-}}
+             ${SSL_KEYFILE:+--ssl-keyfile /etc/ssl/private/server.key}
+             ${SSL_CERTFILE:+--ssl-certfile /etc/ssl/certs/server.crt}
+             ${MODEL_NAME:+--model ${MODEL_NAME-}}
+             ${REVISION:+--revision ${REVISION-}}
+             ${DATATYPE:+--dtype ${DATATYPE-}}
+             ${KVCACHE:+--kv-cache-dtype ${KVCACHE-}}
+             ${CONTEXT_LENGTH:+--max-model-len ${CONTEXT_LENGTH-}}
+             ${NUM_GPUS:+--tensor-parallel-size ${NUM_GPUS-}}
+             ${GPU_MEMORY_UTILIZATION:+-gmu ${GPU_MEMORY_UTILIZATION-}}
+             ${QUANTIZATION:+--quantization ${QUANTIZATION-} --dtype half}
+             ${ENFORCE_EAGER:+--enforce-eager}
+             ${CMD_ADDITIONAL_ARGUMENTS-}
     deploy:
       resources:
         reservations:
           devices:
-          - driver: nvidia
-            count: all
-            capabilities: [gpu]
+            - capabilities: [gpu]
+              count: all
+              driver: nvidia
+    entrypoint: python3 -m aphrodite.endpoints.openai.api_server
```

Comment:
Can we have the entrypoint be configurable as well? We also have the Kobold endpoint at `aphrodite.endpoints.kobold.api_server`, and some users may prefer that because of the embedded UI.

Reply:
Entrypoint is now configurable, with OpenAI being the default.